packages/adminrouter/extra/src/README.md

```diff
@@ -31,6 +31,229 @@ Use `make check-api-docs` to validate that the YAML and HTML files are up to dat
 ## Ports summary
 <img src="admin-router-table.png" alt="" width="100%" align="middle">
 
+## Repository structure
+
+There are two Admin Router "flavours" residing in DC/OS repos:
+  * https://github.com/dcos/dcos/tree/master/packages/adminrouter - Opensource version or `Open` in short
+  * https://github.com/mesosphere/dcos-enterprise/tree/master/packages/adminrouter - Enterprise version or `EE` in short
+
+The `Open` version is the base on top of which `EE` version is built. `EE` is in
+fact an overlay on top of `Open`, it re-uses some of its components.
+
+### Complexity vs. code re-use
+It is crucial to understand that the more generalised the AR repositories
+are, the more complicated they will become. Increased complexity will result in
+people making mistakes and/or situations where complex rules are violated in
+favour of development speed and thus copypasting code. It's all about striking
+the right balance so sometimes code *is* duplicated between repositories in order
+to make it easier for contributors to work with repositories.
+
+### Nginx includes
+All AR code, both Lua and non-Lua, can be divided into following groups:
+ * common code for masters and agents, both EE and Open
+ * agent-specific code, both EE and Open
+ * master-specific code, both EE and Open
+ * Open-specific code, both agent and master
+ * EE-specific code, both agent and master
+ * EE agent specific code
+ * EE master specific code
+ * Open agent specific code
+ * Open master specific code
+on top of that, Nginx-specific configuration is divided into three sections:
+ * main
+ * HTTP
+ * server
+This gives us in total 27 possible "buckets" for Nginx directives. The
```

Comment:
There needs to be a newline between bullet lists and new sections for this to render properly. Else this looks like it is part of the last bullet.